src/intellisense/notebookPythonPathService.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { extensions, Uri } from 'vscode';
+import { IInteractiveWindowProvider } from '../interactive-window/types';
+import { findAssociatedNotebookDocument } from '../notebooks/helpers';
+import { INotebookControllerManager } from '../notebooks/types';
+import { IExtensionSingleActivationService } from '../platform/activation/types';
+import { IPythonApiProvider } from '../platform/api/types';
+import { IVSCodeNotebook } from '../platform/common/application/types';
+import { PylanceExtension, PythonExtension } from '../platform/common/constants';
+import { getFilePath } from '../platform/common/platform/fs-paths';
+import { IConfigurationService } from '../platform/common/types';
+import { IInterpreterService } from '../platform/interpreter/contracts';
+import * as semver from 'semver';
+import { traceInfo, traceVerbose } from '../platform/logging';
+
+/**
+* Manages use of the Python extension's registerJupyterPythonPathFunction API which
+* enables us to provide the python.exe path for a notebook as required for Pylance's
+* LSP-based notebooks support.
+*/
+@injectable()
+export class NotebookPythonPathService implements IExtensionSingleActivationService {
```

Comment:
If you can think of a better name for this service, please let me know. Would `PylancePythonPathService` be better? Or should it include the term `node` since that seems to equate to Pylance in your naming?